driver/src/main/java/org/neo4j/driver/internal/async/connection/BoltProtocolUtil.java

```diff
@@ -42,9 +42,9 @@
 
     private static final ByteBuf HANDSHAKE_BUF = unreleasableBuffer( copyInt(
             BOLT_MAGIC_PREAMBLE,
-            BoltProtocolV43.VERSION.toInt(),
-            BoltProtocolV42.VERSION.toInt(),
+            BoltProtocolV43.VERSION.toIntRange( BoltProtocolV42.VERSION ),
```

Comment:
Then we will lose the multi database support on the version 4.0, since i will negotiate as 3.5. 
Uses 4.1 to talk with a 4.2.x server will not represent any feature drop.

Reply:
Ah good point. I guess we just have to be a little careful whilst we switch to the new handshake

Reply:
I've made a comment on the protocol definition document, to fix the example to consider the version range in decreasing order.

Reply:
Is the idea that we should go to BoltProtocolV43.VERSION.toIntRange(BoltProtocolVersionV40.VERSION)? That way we are saying all version 4 protocols are accepted by this driver and save 2 slots in the handshake.

Reply:
it could be done in the protocol, but I think we should keep using the slots for the 4.0 and 4.1, just in case we connect to a non-update server which do not know how to interpret the extend handshake.